src/docs/5.31.x/core-development-concepts/basics/routes-and-events.mdx

```diff
@@ -0,0 +1,245 @@
+  ]
+})
+```
+
+
+<Alert type="warning">
+  You can define multiple routes per handler, but you can define only a single event per handler.
+</Alert>
+
+## Adding New Routes
+
+Adding new routes is quite simple, but you need to add them via both Pulumi code and the RoutePlugin.
+
+The Pulumi code goes into `apps/api/webiny.application.ts`. You must add the new route there, check out the example where we are adding `[POST]/webiny` route.
+```ts
+import { createApiApp } from "@webiny/serverless-cms-aws";
+import { ApiGraphql } from "@webiny/pulumi-aws";
+
+export default createApiApp({
+    pulumi: app => {
+        const graphQLModule = app.getModule(ApiGraphql);
+
+        graphQLModule.addRoute({
+            // name must be in kebab-case
+            name: "webiny",
+            // path must start with /
+            path: "/webiny",
+            // all http methods allowed + ANY to catch all requests
+            method: "POST"
+        });
+    }
+});
+
+```
+
+Next thing you need to do is to add it into the `apps/api/graphql/src/index.ts` file via the `RoutePlugin` or `createApiGatewayRoute` method:
+
+```ts apps/api/graphql/src/index.ts
+// ... other imports
+import { createApiGatewayHandler as createHandler, createApiGatewayRoute } from "@webiny/handler-aws";
+
+export const handler = createHandler({
+    plugins: [
+        // ... other plugins
+        createApiGatewayRoute(({ onPost }) => {
+            onPost("/webiny", async (request, reply) => {
+                // we can log the whole request body
+                console.log(request.body);
+
+                // and we can send some reply
+                reply
+                    .headers({
+                        "x-route-example": "yes"
+                    })
+                    .send({
+                        everything: {
+                            ok: true
+                        }
+                    });
+            });
+        })
+    ],
+    http: { debug }
+});
+```
+
+<Alert type="warning">
+  Security checks will be done as on the <strong>/graphql</strong> endpoint. If you need customizations, feel free to add them.
+</Alert>
+
+<Alert type="danger">
+  Routes can be added only when initializing the handler.
+</Alert>
+
+## Adding Event Handler
+
+The idea behind all our predefined event handlers is to handle events which are not of API Gateway Event type.
+
+Yes, you can catch API Gateway event like this, but there is no point since we have a handler specifically designed for that.
+Of course, feel free to use what ever is available and can help you to achieve your goal.
+
+
+### Example
+Let's say you created a part of code which sends out an SQS message, and you want to have a Lambda which handles that message.
+
+Good example would be if you want to run some calculation, asynchronously, from our GraphQL Lambda.
+You would insert an SQS Message and in turn it would trigger a Lambda which you have defined. That lambda should have code similar to this:
```

Comment:
```suggestion
You would insert an SQS Message and in turn it would trigger a Lambda which you have defined. 

That Lambda should have code similar to this:
```